azure-pipelines/release.yml

```diff
@@ -15,66 +15,100 @@ parameters:
-      If ("$(resources.pipeline.officialBuildCI.sourceBranch)" -eq "refs/heads/main") {
-        $additionalPublishArgs += "--pre-release"
-        Write-Host "Publish to pre-release channel."
-      } ElseIf ("$(resources.pipeline.officialBuildCI.sourceBranch)" -eq "refs/heads/release") {
-        Write-Host "Publish to release channel."
-      } Else {
-        throw "Unexpected branch name: $(resources.pipeline.officialBuildCI.sourceBranch)."
-      }
-      $additionalPublishArgs += '--packagePath'
+        $additionalPublishArgs = , "publish"
+        # Artifacts are published to either pre-release or release based on the build branch, https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions
+        If ("$(resources.pipeline.officialBuildCI.sourceBranch)" -eq "refs/heads/main") {
+          $additionalPublishArgs += "--pre-release"
+          Write-Host "Publish to pre-release channel."
+        } ElseIf ("$(resources.pipeline.officialBuildCI.sourceBranch)" -eq "refs/heads/release") {
+          Write-Host "Publish to release channel."
+        } Else {
+          throw "Unexpected branch name: $(resources.pipeline.officialBuildCI.sourceBranch)."
+        }
+        $additionalPublishArgs += '--packagePath'
 
-      $platforms = "arm64", "x64", "ia32"
-      $allVsixs = Get-ChildItem $publishArtifacts *.vsix
-      foreach ($vsix in $allVsixs) {
-        foreach ($plat in $platforms) {
-          if ($vsix.Name.Contains($plat)) {
-              $additionalPublishArgs += $vsix
+        $platforms = "arm64", "x64", "ia32"
+        $allVsixs = Get-ChildItem $publishArtifacts *.vsix
+        foreach ($vsix in $allVsixs) {
+          foreach ($plat in $platforms) {
+            if ($vsix.Name.Contains($plat)) {
+                $additionalPublishArgs += $vsix
+            }
           }
         }
-      }
 
-      Write-Host "Command run is: vsce $($additionalPublishArgs)."
-      If ("${{ parameters.test }}" -eq "true") {
-        Write-Host "In test mode, command is printed instead of run."
-        Write-Host "🔒 Verify PAT."
-        vsce verify-pat ms-dotnettools
-      }
-      Else {
-        vsce @additionalPublishArgs
-      }
-    displayName: 🚀 Publish to Marketplace
-    workingDirectory: $(Pipeline.Workspace)
-    env:
-      VSCE_PAT: $(VSCodeMarketplacePAT)
+        Write-Host "Command run is: vsce $($additionalPublishArgs)."
+        If ("${{ parameters.test }}") {
+          Write-Host "In test mode, command is printed instead of run."
+          Write-Host "🔒 Verify PAT."
+          vsce verify-pat ms-dotnettools
+        }
+        Else {
+          vsce @additionalPublishArgs
+        }
+      displayName: 🚀 Publish to Marketplace
+      workingDirectory: $(Pipeline.Workspace)
+      env:
+        VSCE_PAT: $(VSCodeMarketplacePAT)
+- stage: 'TagReleaseNumberToRoslyn'
+  displayName: 'Tag release number of vscode-csharp to Roslyn'
+  dependsOn: 'PublishStage'
```

Comment:
![image](https://github.com/dotnet/vscode-csharp/assets/24360909/cd09b84e-bbe4-42a3-825e-c968d70cff3d)
The doc said it will run if the prev steps/job succeed, but here it depends on the stage. To be safe I added the `succeeded('PublishStage')`